paddle/fluid/pir/transforms/pd_op_to_kernel_pass.cc

```diff
@@ -417,54 +338,73 @@ pir::OpResult AddDtypeTransferOp(pir::Value in,
   pir::OpResult new_in = op->result(0);
   return new_in;
 }
-pir::Type BuildOutputType(pir::Type type,
-                          const phi::Place& place,
-                          phi::DataType data_type,
-                          pir::IrContext* ctx) {
-  if (type.isa<dialect::DenseTensorType>()) {
-    auto dense_tensor_type = type.dyn_cast<dialect::DenseTensorType>();
-    auto out_dtype = dense_tensor_type.dtype();
-
-    // TODO(phlrain): open this after fix pr(55509) confict
-    // if (data_type != phi::DataType::UNDEFINED) {
-    //   out_dtype = TransToIrDataType(data_type, ctx);
-    // }
-
-    return dialect::AllocatedDenseTensorType::get(
-        ctx,
-        place,
-        out_dtype,
-        dense_tensor_type.dims(),
-        dense_tensor_type.data_layout(),
-        dense_tensor_type.lod(),
-        dense_tensor_type.offset());
-
-  } else if (type.isa<dialect::SelectedRowsType>()) {
-    auto selected_rows_type = type.dyn_cast<dialect::SelectedRowsType>();
-    auto out_dtype = selected_rows_type.dtype();
-
-    // TODO(phlrain): open this after fix pr(55509) confict
-    // if (data_type != phi::DataType::UNDEFINED) {
-    //   out_dtype = TransToIrDataType(data_type, ctx);
-    // }
-    return dialect::AllocatedSelectedRowsType::get(
-        ctx,
-        place,
-        out_dtype,
-        selected_rows_type.dims(),
-        selected_rows_type.data_layout(),
-        selected_rows_type.lod(),
-        selected_rows_type.offset());
+
+template <class T>
```

Comment:
```suggestion
template <class IrType>
```
这里看起来也是IrType? 可以统一下，可读性更好一些？

Reply:
done, thanks!